### PR TITLE
Replace flake8 and black with ruff

### PR DESCRIPTION
**Pull request by Spacetown**
Replace the tools and configure them in `pyproject.toml`. Move also the configuration for purest.

### CODE REVIEW

File: gcovr/formats/lcov/write.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def postfix():
             for lineno in sorted_lines:
                 line_coverage = covdata[key].lines[lineno]
                 if line_coverage.excluded:
-                    next
+                    continue
```
Comment (latk) [inline gcovr/formats/lcov/write.py:95]: Oh wow what an excellent find!

File: noxfile.py [hunk 2]
```diff
@@ -270,49 +266,42 @@ def qa(session: nox.Session) -> None:
...
-            f"Skip black because of platform {platform.python_implementation()}."
-        )
-
 
 @nox.session
-def flake8(session: nox.Session) -> None:
-    """Run flake8."""
-    session.install("flake8", "flake8-print")
+def ruff_check(session: nox.Session) -> None:
+    """Run bandit, a code formatter and format checker."""
```
Comment (latk) [inline noxfile.py:276]: Docstring incorrectly references "bandit"? Seems like after a couple rounds of changes, many docstrings have become out of date.

File: pyproject.toml [hunk 3]
```diff
@@ -1,3 +1,46 @@
...
+unfixable = ["B"]
+
+# 4. Ignore `E402` (import violations) in all `__init__.py` files, and in selected subdirectories.
+[tool.ruff.lint.per-file-ignores]
+"__init__.py" = ["E402"]
+"**/{tests,docs,tools}/*" = ["E402"]
+
+[tool.ruff.format]
+# 5. Use single quotes in `ruff format`.
+quote-style = "double"
```
Comment (latk) [inline pyproject.toml:29]: Docstring contradicts the configuration value.

**Review by latk — APPROVED** [review]
Thank you for working towards pyproject.toml-ification!

File: gcovr/formats/lcov/write.py (hunk 1 above)
Comment (Spacetown) [reply gcovr/formats/lcov/write.py:95]: I was also wondering about this.

**Comment by Spacetown** [comment]
As next step I want to check on pylon and on hatchling.

**Pull request merged**